tests/conftest.py

```diff
@@ -37,18 +37,17 @@ def local_pulumi_login():
     run([pulumi_path, "logout"], check=False)
 
 
-@fixture(autouse=True)
-def log_directory(monkeypatch, tmp_path):
-    monkeypatch.setenv("DSH_LOG_DIRECTORY", tmp_path)
-    init_logging()
-    return tmp_path
-
-
 @fixture(autouse=True, scope="session")
-def log_file(session_mocker):
+def log_directory(session_mocker, tmp_path_factory):
     session_mocker.patch.object(
         data_safe_haven.logging.logger, "logfile_name", return_value="test.log"
     )
+    log_dir = tmp_path_factory.mktemp("logs")
+    session_mocker.patch.object(
+        data_safe_haven.directories, "log_dir", return_value=log_dir
+    )
+    init_logging()
+    return log_dir
```

Comment:
:D I just came up with the same potential solution. Does it work?

Reply:
No - this function isn't getting called by any tests that don't explicitly call it.

Reply:
I think so 🤞 :smile: